packages/validation/src/isAsync.js

```diff
@@ -0,0 +1,6 @@
+import isPromise from './isPromise';
+import isFunction from './isFunction';
+
+export default function isAsync(value) {
+    return isPromise(value) || (isFunction(value) && value() instanceof Promise);
+}
```

Comment:
why execute the function to know if it's async, what happens if executing the function cause undesired results :/

we want to know if the function uses async/await
```js
return value.constructor.name === "AsyncFunction";
```

Reply:
this are the tests results with return value.constructor.name === "AsyncFunction"; method...
async function () {} -> false
async () => {} -> false,
new Promise(() => {})  -> false... // this can be fixed with isPromise check

Reply:
I understand but if I don't want to execute the function in any way ?? what if I'm checking if is async then execute another function

Reply:
checking if method is async is quite complicated... i found another solution that works without executing the function... i don't like it... it is based on string matching in the code generated by babel... code that  may change from one version to another... but it works

every function with `async` keyword is translated to the same code while normal functions  keep their original code
```js
console.log((async function () {}).toString().trim());
```
outputs:
```js
function () {
    var self = this,
    args = arguments;
    return new Promise(function (resolve, reject) 
        var gen = fn.apply(self, args);
        function _next(value) {
              asyncGeneratorStep(gen, resolve, reject, _next, _throw, "next", value);
        }
        function _throw(err) {
            asyncGeneratorStep(gen, resolve, reject, _next, _throw, "throw", err);
        }
        _next(undefined);
    });
}
```